daemons/ipa-slapi-plugins/ipa-extdom-extop/ipa_extdom_common.c

```diff
@@ -97,25 +98,48 @@ static int inc_buffer(size_t buf_max, size_t *_buf_len, char **_buf)
     return 0;
 }
 
-int getpwnam_r_wrapper(size_t buf_max, const char *name,
+int __nss_to_err(enum nss_status errcode)
+{
+    int ret = -1;
+
+    if (errcode == NSS_STATUS_SUCCESS) {
+        ret = 0;
+    } else if (errcode == NSS_STATUS_NOTFOUND) {
+        ret = ENOENT;
+    } else if(errcode == NSS_STATUS_TRYAGAIN) {
+        ret = ERANGE;
+    } else if(errcode == NSS_STATUS_UNAVAIL) {
+        ret = ETIMEDOUT;
+    };
+
+    return ret;
+}
+
+int getpwnam_r_wrapper(struct ipa_extdom_ctx *ctx, const char *name,
                        struct passwd *pwd, char **_buf, size_t *_buf_len)
 {
     char *buf = NULL;
     size_t buf_len = 0;
-    int ret;
+    int ret, lerrno = 0;
     struct passwd *result = NULL;
+    enum nss_status rc;
 
     buf = *_buf;
     buf_len = *_buf_len;
 
-    while (buf != NULL
-            && (ret = getpwnam_r(name, pwd, buf, buf_len, &result)) == ERANGE) {
-        ret = inc_buffer(buf_max, &buf_len, &buf);
-        if (ret != 0) {
-            if (ret == ERANGE) {
-                LOG("Buffer too small, increase ipaExtdomMaxNssBufSize.\n");
+    while (buf != NULL) {
```

Comment:
while you are changing this, improve it this way:
for (buf = *_buf, buf_len = *_buf_len; buf != NULL;) {